# HG changeset patch
# User xunxun1982
# Date 1411495437 -28800
#      Wed Sep 24 02:03:57 2014 +0800
# Branch MOBILE400b10_2015080313_RELBRANCH
# Node ID 0c323daed882838d08bfad1d1479f6a384306c6c
# Parent  677c3e77d58b79e3a58a85918f021726fef33f76
trim-mini

diff --git a/widget/windows/nsWindow.cpp b/widget/windows/nsWindow.cpp
--- a/widget/windows/nsWindow.cpp
+++ b/widget/windows/nsWindow.cpp
@@ -641,17 +641,17 @@ nsWindow::Create(nsIWidget *aParent,
   // do some initialization work.
   if (sTrimOnMinimize == 2 && mWindowType == eWindowType_invisible) {
     // Our internal trim prevention logic is effective on 2K/XP at maintaining
     // the working set when windows are minimized, but on Vista and up it has
     // little to no effect. Since this feature has been the source of numerous
     // bugs over the years, disable it (sTrimOnMinimize=1) on Vista and up.
     sTrimOnMinimize =
       Preferences::GetBool("config.trim_on_minimize",
-        IsVistaOrLater() ? 1 : 0);
+        false);
     sSwitchKeyboardLayout =
       Preferences::GetBool("intl.keyboard.per_window_layout", false);
   }
 
   // Query for command button metric data for rendering the titlebar. We
   // only do this once on the first window.
   if (mWindowType == eWindowType_toplevel &&
       (!nsUXThemeData::sTitlebarInfoPopulatedThemed ||
@@ -1668,16 +1668,21 @@ NS_IMETHODIMP nsWindow::SetSizeMode(int3
       case nsSizeMode_Minimized :
         // Using SW_SHOWMINIMIZED prevents the working set from being trimmed but
         // keeps the window active in the tray. So after the window is minimized,
         // windows will fire WM_WINDOWPOSCHANGED (OnWindowPosChanged) at which point
         // we will do some additional processing to get the active window set right.
         // If sTrimOnMinimize is set, we let windows handle minimization normally
         // using SW_MINIMIZE.
         mode = sTrimOnMinimize ? SW_MINIMIZE : SW_SHOWMINIMIZED;
+		if (sTrimOnMinimize) {
+			HANDLE hProcess = GetCurrentProcess();
+			if(hProcess != NULL)
+				SetProcessWorkingSetSize(hProcess, -1, -1);
+		}
         break;
 
       default :
         mode = SW_RESTORE;
     }
 
     WINDOWPLACEMENT pl;
     pl.length = sizeof(pl);
